### PR TITLE
Prevent root_count overflow in GcHeader

### PR DESCRIPTION
**Pull request by mrhapile**
### Summary

This PR prevents a potential overflow in `GcHeader::inc_roots`.

`root_count` is currently stored as `Cell<u16>`. The previous implementation incremented the counter using unchecked arithmetic:
Screenshot 2026-03-05 at 6 05 36 PM
Screenshot 2026-03-05 at 6 07 31 PM

```rust
self.root_count.set(self.root_count.get() + 1);
```

If more than `u16::MAX` roots were created:

* **Debug builds** panic due to overflow checks
* **Release builds** silently wrap the value to `0`

Screenshot 2026-03-05 at 6 09 08 PM
Screenshot 2026-03-05 at 6 17 24 PM

Silent wraparound violates the garbage collector invariant that:

```
root_count == 0 → object can be collected
```

If the counter wraps from `65535 → 0`, a live object could incorrectly appear unreferenced and become eligible for collection.

---

### Changes

* Replace unchecked increment with `checked_add`
* Add explanatory safety comment in `inc_roots`
* Add a regression test to ensure overflow cannot silently occur

Updated implementation:

```rust
pub fn inc_roots(&self) {
    // Prevent silent overflow of root_count.
    // In release builds, `u16::MAX + 1` would wrap to 0 and break GC invariants.
    let current = self.root_count.get();
    let new = current
        .checked_add(1)
        .expect("GcHeader root_count overflow");

    self.root_count.set(new);
}
```

---

### Tests

Added a regression test ensuring overflow is detected:

```rust
#[test]
#[should_panic(expected = "GcHeader root_count overflow")]
fn root_count_overflow_panics() {
    let header = GcHeader::new_white();

    for _ in 0..=u16::MAX {
        header.inc_roots();
    }
}
```

All existing tests pass successfully.
Screenshot 2026-03-05 at 6 43 25 PM

---

### Impact

* Prevents silent integer wraparound in release builds
* Preserves existing `u16` storage layout
* Maintains `no_std` compatibility
* Ensures GC invariants remain valid

### CODE REVIEW

File: oscars/src/collectors/mark_sweep/tests.rs [hunk 1]
```diff
@@ -1,3 +1,6 @@
+#[cfg(feature = "std")]
```
Comment (nekevss) [inline oscars/src/collectors/mark_sweep/tests.rs]: issue: This should be unneeded. Please remove.

(I enabled std in test modules, so you should be okay if you rebase to main)
